catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/delegate/SearchTerm.java

```diff
@@ -47,7 +51,12 @@ public boolean match(String other) {
       return true;
     }
     if (pattern != null) {
-      return pattern.matcher(other).matches();
+      try {
+        return TimeLimitedMatcher.create(pattern, other).matches();
```

Comment:
I am curious what the time limit is set to by default

Reply:
Five seconds. Way longer than should be necessary for a regex search normally. See #3699 for the TLMatcher.